### PR TITLE
[5.5] Add an "all" method to the query builder

### PR DESCRIPTION
**Pull request by JosephSilber**

### CODE REVIEW

**Comment by ConnorVG** [comment]
`all` makes it look like you're going to always get _all_ of a specific model. You are **not**. This makes it semantically bizarre in many cases (if we were to use `all`).

**Comment by JosephSilber** [comment]
@ConnorVG you would expect `User:where('active', true)->all()` to give you back _all_ users???

**Comment by ntzm** [comment]
Won't this break code like `User::where('x', 'y')->all()`? Obviously the code is wrong but I see some people doing it accidentally.

**Comment by ConnorVG** [comment]
@JosephSilber you don't always have the query building in scope, just the resulting builder. `all` is semantically equivalent to `everything`. So… yeah, it's weird to add if that's not what it does.

**Comment by JosephSilber** [comment]
@ntzm that's actually the point of this PR; so that you don't get that error.

**Comment by ntzm** [comment]
So it's a breaking change? Should be targeting master then

**Comment by JosephSilber** [comment]
@ntzm it's not a breaking change :see_no_evil:

**Comment by sixlive** [comment]
Always forget to add `get()` onto the end of these. This sounds great!

**Comment by deleugpn** [comment]
It's only a breaking change if you write code that calls a method not defined (`all`) and then catches the exception to do something :joy:

**Comment by Dylan-DPC** [comment]
>  Always forget to add get() onto the end of these. This sounds great!
@sixlive how is that great? now you will forget to do `all()` :laughing:

File: src/Illuminate/Database/Concerns/BuildsQueries.php [hunk 1]
```diff
@@ -9,6 +9,16 @@
 trait BuildsQueries
 {
     /**
+     * Execute the query, while selecting all columns.
+     *
+     * @return \Illuminate\Support\Collection
+     */
+    public function all()
+    {
+        $this->get();
```
Comment (bonzai) [inline src/Illuminate/Database/Concerns/BuildsQueries.php]: Missing return statement.

**Comment by taylorotwell** [comment]
We might want to know how the tests passed for this with no return statement

**Comment by ConnorVG** [comment]
@taylorotwell the current tests passing is because the results aren't utilised in that specific (altered) test, the return is not needed for that case.

That said, this PR adds no tests, which is **very** disappointing.

**Comment by deleugpn** [comment]
The tests passed because it's only using `all` to test the nullable (empty) return.

**Comment by linaspasv** [comment]
This PR is just wrong... all keyword stands for everything, and should not be used in any other context.

**Comment by devcircus** [comment]
The context is the caller. In this case, 'all' makes perfect sense.

**Comment by linaspasv** [comment]
Well, as stated in the documentation all is being used to retrieve every record, and its the only usage of the keyword. Not really sure if it is smart to reinvent something here, and change or alias  get to something that just naturally feels wrong. If people do mistakes its really not the fault of the framework but lack of focus and attention to defined rules. Personally I have never experienced any problems in using get.

**Comment by taylorotwell** [comment]
I guess I'll close this based on negative feedback.

**Pull request closed without merging**